src/parser/meta.rs

```diff
@@ -0,0 +1,813 @@
+use nom::{
+    branch::alt,
+    bytes::streaming::{tag, take, take_while1},
+    character::{
+        complete::digit1,
+        streaming::{crlf, space0, space1},
+    },
+    combinator::{map, map_res, value},
+    multi::many0,
+    sequence::{pair, preceded, terminated, tuple},
+    IResult,
+};
+
+use super::{parse_u32, ErrorKind, MetaValue, Response, Status, Value};
+
+// TODO: remove this later in favour of individual parse_meta_*_status methods
+#[allow(dead_code)]
+pub fn parse_meta_status(buf: &[u8]) -> IResult<&[u8], Response> {
+    terminated(
+        alt((
+            value(Response::Status(Status::NotStored), tag(b"NS")),
+            value(Response::Status(Status::Deleted), tag(b"DE")),
+            value(Response::Status(Status::Touched), tag(b"TO")),
+            value(Response::Status(Status::Exists), tag(b"EX")),
+            value(Response::Status(Status::NotFound), tag(b"NF")),
+        )),
+        crlf,
+    )(buf)
+}
+
+pub fn parse_meta_set_status(buf: &[u8]) -> IResult<&[u8], Response> {
+    alt((
+        value(Response::Status(Status::Stored), tag(b"HD")),
+        value(Response::Status(Status::NotStored), tag(b"NS")),
+        value(Response::Status(Status::Exists), tag(b"EX")),
+        value(Response::Status(Status::NotFound), tag(b"NF")),
+    ))(buf)
+}
+
+pub fn parse_meta_get_status(buf: &[u8]) -> IResult<&[u8], Response> {
+    alt((
+        value(Response::Status(Status::Value), tag(b"VA ")),
```

Comment:
Nuance of now `nom` works.  The ["VA" response](https://github.com/memcached/memcached/blob/5609673ed29db98a377749fab469fe80777de8fd/doc/protocol.txt#L519) (cache hit with value requested / `v` flag used) will always have a space following it so we can parse all 3 characters in one shot in this case, instead of parsing the two character code and then parsing the space to get to the value size.

Example `VA` response:
```
VA <size> <flags>*\r\n.    => VA always followed by whitespace and <size>
<data block>\r\n
```

Alternate response on cache hit with no value requested / `v` flag not used yields `HD` response code which can take these forms:

```
HD <flags>*\r\n    => any non-v flags used, space follows HD

or

HD\r\n     => no flags used, no space following HD, straight into carriage return
```